Index: Platform/IWindowInfo.cs
===================================================================
--- Platform/IWindowInfo.cs	(revision 3123)
+++ Platform/IWindowInfo.cs	(working copy)
@@ -15,5 +15,9 @@
     /// <summary>Descibes an OS window.</summary>
     public interface IWindowInfo : IDisposable
     {
+        /// <summary>
+        /// returns an IntPtr to the NativeWindow
+        /// </summary>
+        IntPtr WindowHandle { get; }
     }
 }
Index: Platform/Egl/EglWindowInfo.cs
===================================================================
--- Platform/Egl/EglWindowInfo.cs	(revision 3123)
+++ Platform/Egl/EglWindowInfo.cs	(working copy)
@@ -129,5 +129,20 @@
         }
 
         #endregion
+
+        #region Implementation of IWindowInfo
+
+        /// <summary>
+        /// returns an IntPtr to the NativeWindow
+        /// </summary>
+        public IntPtr WindowHandle
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+        #endregion
     }
 }
Index: Platform/Dummy/DummyWindowInfo.cs
===================================================================
--- Platform/Dummy/DummyWindowInfo.cs	(revision 3123)
+++ Platform/Dummy/DummyWindowInfo.cs	(working copy)
@@ -13,5 +13,20 @@
         }
 
         #endregion
+
+        #region Implementation of IWindowInfo
+
+        /// <summary>
+        /// returns an IntPtr to the NativeWindow
+        /// </summary>
+        public IntPtr WindowHandle
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        #endregion
     }
 }
Index: Platform/MacOS/CarbonWindowInfo.cs
===================================================================
--- Platform/MacOS/CarbonWindowInfo.cs	(revision 3123)
+++ Platform/MacOS/CarbonWindowInfo.cs	(working copy)
@@ -134,5 +134,20 @@
         }
         
         #endregion
+
+        #region Implementation of IWindowInfo
+
+        /// <summary>
+        /// returns an IntPtr to the NativeWindow
+        /// </summary>
+        public IntPtr WindowHandle
+        {
+            get
+            {
+                return windowRef;
+            }
+        }
+
+        #endregion
     }
 }
 